File: typescript-client-sdk/src/sdk/models/operations/getsubscriptionssubscriptionidusage.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { Transform, Type } from "class-transformer";


export class GetSubscriptionsSubscriptionIdUsagePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=subscription_id" })
  subscriptionId: string;
}
export enum GetSubscriptionsSubscriptionIdUsageGranularityEnum {
    Day = "day"
}
export enum GetSubscriptionsSubscriptionIdUsageViewModeEnum {
    Periodic = "periodic",
    Cumulative = "cumulative"
}

export class GetSubscriptionsSubscriptionIdUsageQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=billable_metric_id" })
  billableMetricId?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=granularity" })
  granularity?: GetSubscriptionsSubscriptionIdUsageGranularityEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=group_by" })
  groupBy?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=timeframe_end;dateTimeFormat=YYYY-MM-DDThh:mm:ss.sssZ" })
  timeframeEnd?: Date;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=timeframe_start;dateTimeFormat=YYYY-MM-DDThh:mm:ss.sssZ" })
  timeframeStart?: Date;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=view_mode" })
  viewMode?: GetSubscriptionsSubscriptionIdUsageViewModeEnum;
}

export class GetSubscriptionsSubscriptionIdUsageRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetSubscriptionsSubscriptionIdUsagePathParams;

  @SpeakeasyMetadata()
  queryParams: GetSubscriptionsSubscriptionIdUsageQueryParams;
}

export class GetSubscriptionsSubscriptionIdUsageResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getsubscriptionssubscriptionidusage.go
package operations

import (
	"time"
)

type GetSubscriptionsSubscriptionIDUsagePathParams struct {
	SubscriptionID string `pathParam:"style=simple,explode=false,name=subscription_id"`
}

type GetSubscriptionsSubscriptionIDUsageGranularityEnum string

const (
	GetSubscriptionsSubscriptionIDUsageGranularityEnumDay GetSubscriptionsSubscriptionIDUsageGranularityEnum = "day"
)

type GetSubscriptionsSubscriptionIDUsageViewModeEnum string

const (
	GetSubscriptionsSubscriptionIDUsageViewModeEnumPeriodic   GetSubscriptionsSubscriptionIDUsageViewModeEnum = "periodic"
	GetSubscriptionsSubscriptionIDUsageViewModeEnumCumulative GetSubscriptionsSubscriptionIDUsageViewModeEnum = "cumulative"
)

type GetSubscriptionsSubscriptionIDUsageQueryParams struct {
	BillableMetricID *string                                             `queryParam:"style=form,explode=true,name=billable_metric_id"`
	Granularity      *GetSubscriptionsSubscriptionIDUsageGranularityEnum `queryParam:"style=form,explode=true,name=granularity"`
	GroupBy          *string                                             `queryParam:"style=form,explode=true,name=group_by"`
	TimeframeEnd     *time.Time                                          `queryParam:"style=form,explode=true,name=timeframe_end"`
	TimeframeStart   *time.Time                                          `queryParam:"style=form,explode=true,name=timeframe_start"`
	ViewMode         *GetSubscriptionsSubscriptionIDUsageViewModeEnum    `queryParam:"style=form,explode=true,name=view_mode"`
}

type GetSubscriptionsSubscriptionIDUsageRequest struct {
	PathParams  GetSubscriptionsSubscriptionIDUsagePathParams
	QueryParams GetSubscriptionsSubscriptionIDUsageQueryParams
}

type GetSubscriptionsSubscriptionIDUsageResponse struct {
	ContentType string
	StatusCode  int
}
